bin/templates/cordova/lib/prepare.js

```diff
@@ -305,45 +304,35 @@ function createResourceMap (cordovaProject, locations, resources) {
         case 'customIcon':
             // Copy icon for the App
             targetPath = path.join(locations.www, 'img', `app${resource.extension}`);
-            resourceMap.push(mapResources(cordovaProject.root, resource.src, targetPath));
+            resourceMap.push({ [resource.src]: targetPath });
             // Copy icon for the Installer
             targetPath = path.join(locations.buildRes, `installer${resource.extension}`);
-            resourceMap.push(mapResources(cordovaProject.root, resource.src, targetPath));
+            resourceMap.push({ [resource.src]: targetPath });
             break;
         case 'appIcon':
             targetPath = path.join(locations.www, 'img', `app${resource.extension}`);
-            resourceMap.push(mapResources(cordovaProject.root, resource.src, targetPath));
+            resourceMap.push({ [resource.src]: targetPath });
             break;
         case 'installerIcon':
             targetPath = path.join(locations.buildRes, `installer${resource.extension}`);
-            resourceMap.push(mapResources(cordovaProject.root, resource.src, targetPath));
+            resourceMap.push({ [resource.src]: targetPath });
             break;
         case 'highResIcons':
             for (const key in resource) {
                 const highResIcon = resource[key];
                 targetPath = path.join(locations.www, 'img', 'icon');
                 targetPath += highResIcon.suffix === '1x' ? highResIcon.extension : `@${highResIcon.suffix}${highResIcon.extension}`;
-                resourceMap.push(mapResources(cordovaProject.root, highResIcon.src, targetPath));
+                resourceMap.push({ [highResIcon.src]: targetPath });
             }
             break;
         case 'splashScreen':
             targetPath = path.join(locations.www, `.cdv`, `splashScreen${resource.extension}`);
-            resourceMap.push(mapResources(cordovaProject.root, resource.src, targetPath));
+            resourceMap.push({ [resource.src]: targetPath });
             break;
         }
     }
-    return resourceMap;
-}
 
-/**
- * Get a map containing resources of a specified name (or directory) to the target directory.
- */
-function mapResources (rootDir, sourcePath, targetPath) {
-    const pathMap = {};
-    shell.ls(path.join(rootDir, sourcePath)).forEach(() => {
-        pathMap[sourcePath] = targetPath;
-    });
-    return pathMap;
```

Comment:
We cannot simply remove this, can we? `mapResources` returned `{ [sourcePath]: targetPath }` for files that existed and `{}` for those that didn't. The new logic behaves as if all files existed. A possible implementation without `shelljs` could be:
```js
return fs.existsFileSync(path.join(rootDir, sourcePath))
    ? { [sourcePath]: targetPath } : {};
```

Reply:
I still want to drop the method. 
I added `fs.existsSync` where this method use to be called and added back `{}` though it shouldn't be there. I will be refactoring this area more in a separate PR.